src/inspect/inspect.action.ts

```diff
@@ -0,0 +1,91 @@
+import { Yagna } from "@golem-sdk/golem-js";
+import { InspectOptions } from "./inspect.options";
+import chalk from "chalk";
+import _ from "lodash";
+
+async function getDemandDetails(yagna: Yagna, demandId: string) {
+  const api = yagna.getApi().market;
+  const { data: activeDemands } = await api.getDemands();
+  const demandDetails = activeDemands.find((demand) => demand.demandId === demandId);
+  if (!demandDetails) {
+    throw new Error("Demand not found");
+  }
+  return demandDetails;
+}
+
+async function getAgreementDetails(yagna: Yagna, agreementId: string) {
+  const api = yagna.getApi().market;
+  const { data: agreementDetails } = await api.getAgreement(agreementId);
+  // agreementDetails already has demand details included
+  return agreementDetails;
+}
+
+async function getActivityDetails(yagna: Yagna, activityId: string) {
+  const api = yagna.getApi().activity.state;
+  const {
+    data: { state },
+  } = await api.getActivityState(activityId);
+  const { data: usage } = await api.getActivityUsage(activityId);
+  const agreementId = await api.getActivityAgreementId(activityId);
+  return {
+    currentState: state[0],
+    nextState: state[1],
+    usage,
+    agreement: await getAgreementDetails(yagna, agreementId),
+  };
+}
+
+function printSelectedColumns(obj: object, columns: string[]) {
+  if (columns[0] === "*") {
```

Comment:
For me (macos) I have to use `--columns "*"`. Using `--column *` returns `{}`. Maybe it's worth to validate it.. ?

Reply:
This is a feature of the shell you're using, if you don't wrap * in quotation marks it gets replaced with all files in the current dir (try `echo *`), there is nothing we can do about that from the cli perspective :/